Issue 48 - Sonarqube connection & Code Coverage

Bishares opened the pull request:
The **backend** and **frontend** are now hooked up to sonarqube. resolves #48 
The frontend and backend to these steps
1. Execute tests
2. Generate Code Coverage into coverage.xml (backend) and sonar-report.xml (frontend)
3. Those coverage files are declared in the sonar-project.properties files in backend and frontend
4. sonarqube github action is executed and sonarqube evaluates everything at the running sonarqube instance

Bishares committed:
added sonarqube trigger to ci/cd frontend

Bishares committed:
Update node.js.yml

Bishares committed:
Update node.js.yml

Commit:
Merge branch 'dev' of github.com:h3ssto/ddueruem-web into issue-48

Commit:
added test:ci yarn task

Bishares committed:
Update node.js.yml

Commit:
adjusted jestProcessor path

Commit:
Merge branch 'issue-48' of github.com:h3ssto/ddueruem-web into issue-48

Commit:
added sonar properties configuration file

Bishares committed:
added artifact upload/download of frontend tests

Bishares committed:
Update node.js.yml

Bishares committed:
Update node.js.yml

Commit:
added test filter for sonarqube properties

Commit:
Merge branch 'issue-48' of github.com:h3ssto/ddueruem-web into issue-48

Commit:
added coverage parameters

Bishares committed:
Update node.js.yml

Commit:
corrected sonarqube properties

Bishares committed:
added sonarqube to backend

Commit:
added sonar properties for backend

Commit:
modified backend sonarqube properties

Bishares committed:
Update django.yml

Bishares committed:
fixed env upload/download

Commit:
added django-coverage-plugin to requirements

Bishares committed:
added coverage step for backend

Commit:
added debug:true for django (to enable test coverage)

Commit:
modified backend coverage

Bishares committed:
Update django.yml

Commit:
modified sonarqube backend properties

Commit:
Merge branch 'issue-48' of github.com:h3ssto/ddueruem-web into issue-48

Commit:
modified backend sonarqube properties

Commit:
modified frontend sonar properties

Bishares committed:
Update node.js.yml

Commit:
resolved #48

wurstbroteater commented on backend/.gitignore:
```diff
@@ -0,0 +1,5 @@
+htmlcov/
```
do we need this ignore file here ? I think the idea was to have one project.gitignore to avoid problems. Please check if it is possible to move this to the project .gitignore file

Bishares replied:
everything is now in one gitignore

wurstbroteater commented on backend/ddueruemweb/settings.py:
```diff
@@ -69,6 +69,7 @@
         ,
         'APP_DIRS': True,
         'OPTIONS': {
+            'debug': True,
```
debug in "production" ? Might be discussible.

Bishares replied:
add settings_dev.py

Bishares replied:
So we probably need a script in the production and dev pipeline to set the DJANGO_SETTINGS_MODULE
as described in here
https://stackoverflow.com/a/10664412/12263454

Bishares replied:
I added a new variable TEMPLATE_DEBUG into the env files. The settings.py file is also modified to use this variable

wurstbroteater reviewed (commented):
looks good in general. There are a few things we can resolve in the meeting today

Commit:
- removed .gitignore in backend and added it to the global gitignore
- added TEMPLATE_DEBUG to settings.py and env files (used for Django test coverage command)
resolves #48

wurstbroteater reviewed (approved):
ok
